activemq-broker/src/main/java/org/apache/activemq/broker/BrokerService.java

```diff
@@ -1768,7 +1767,7 @@ public synchronized PListStore getTempDataStore() {
 
             try {
                 PersistenceAdapter pa = getPersistenceAdapter();
-                if( pa!=null && pa instanceof PListStore) {
+                if(pa instanceof PListStore) {
```

Comment:
Why is the null check being removed?

Reply:
Because `instanceof` already handles null checking. I.E if pa == null, instanceof will return false. So In this case, I believe the null checking is redundant. Do you see any concerns?